subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -1461,6 +1461,7 @@ int lwm2m_rd_client_stop(struct lwm2m_ctx *client_ctx,
 
 	client.ctx->event_cb = event_cb;
 	rd_client_message_free();
+	lwm2m_engine_context_close(client_ctx);
```

Comment:
Context will be closed in `ENGINE_DEREGISTERED` state handler (which, depending on the `deregister` parameter is either entered immediately, or after the deregitration procedure), why would it be needed to do it here?

Reply:
If device is more connected, deregistration will fail and block the closing process?
I can remove it from this PR: maybe out of scope. Please let me know.